### DISCUSSION TITLE
SUPER WEIRD ISSUE

### DISCUSSION THREAD

**Discussion by matthewhisle** [ujjwalguptaofficial/JsStore#383] [General]
I have a SUPER WEIRD issue.

It is saying no index exists on a column only if I don't capitalize the W in where. The field does exist on the table.

![image](https://github.com/user-attachments/assets/09d28cd1-1c24-447c-898c-2f78b96326f8)

**Comment by matthewhisle** [comment]
This is a new field that was added using the schema modifier by raising the version number by one. Also curiously, none of the new fiedls I am adding with version numbers, while they do show in the tables, they do not show when viewing in chrome indexeddb.

**Comment by matthewhisle** [comment]
Yeah it doesnt seem like where clauses work properly on any field that is added using the schema version number changing system

SUPER WEIRDLY it does work if you capitalize the W in WHERE...

**Comment by matthewhisle** [comment]
It appears like it has something to do like such

I had a device on version 13, I added this particular field for version 14... The device upgraded, and acquired that field

Other devices had never seen this database before at all... So they were starting from nothing... They did NOT receive this field?

Do we need to add new fields to both the base schema AND the alter section perhaps? The documentation has new fields showing only in the alter section, but not added to the base schema for the table

**Reply by ujjwalguptaofficial** [reply]
Hey @matthewhisle, i will check this and let you know. I guess this might be the condition where the olderschema and newerschema are not properly working in newer devices where db is created for first time. There might be some edge cases. 

I will check this on priority and let you know.

**Reply by matthewhisle** [reply]
Thank you so much for the help! My theory is that when adding a new column to the table, we may need to add it to the base schema AND the "Alter -> Add" schema? Just let me know what you think once you are able to take a look at it. I use this pretty heavily, but just realized it was like this tonight, when I had a device that was working fine, I cleared all chrome data, it initialized database brand new, then it suddenly started saying it was missing index to use field for where clause.

**Reply by ujjwalguptaofficial** [reply]
gotcha, will update you.

**Reply by ujjwalguptaofficial** [reply]
I confirm this is the issue, while this works in the same device but does not work in new device. Initial assumption was - that new schema changes will be also defined in the base schema, but it's clearly wrong and does not make sense.

i am working on this currently and fix should be rolled by EOD.

**Comment by ujjwalguptaofficial** [comment]
@matthewhisle this is fixed in version - 4.8.3. Thanks for reporting this and explaining. We have also added test cases considering this specific case, to make sure this does not happen again.

Let me know if you still see this or something else.

**Reply by matthewhisle** [reply]
No problem at all! Thank you so much. Just out of curiosity, if I had been adding the new fields to the base schema AND the alter -> add section, instead of putting it only in the alter -> add section, would that have worked for both cases and not caused any issues?

**Reply by matthewhisle** [reply]
Also is the 4.8.3 version on :

https://cdn.jsdelivr.net/npm/jsstore/dist/jsstore.min.js
https://cdn.jsdelivr.net/npm/jsstore/dist/jsstore.worker.min.js

Just wanted to make sure. These are usually where I get the code from

**Reply by matthewhisle** [reply]
Ignore previous message.

So I upgraded to version 4.8.3, but it seems to still not add the fields when creating the database for the first time. Sorry if I am doing something wrong :( Just to confirm, I don't need to add these fields to the base schema too correct?

**Reply by matthewhisle** [reply]
So I did some testing. It looks like even on the old version it works good if:

Any column you add to the (Alter -> Add Section)

You ALSO add to the base schema

**Reply by ujjwalguptaofficial** [reply]
Yes old version also works if you add this in base schema, but with this fix you dont need to.

I think you should just upgrade the version to +1 , this will trigger the database creation and thus reindexing everything from alter section.

**Reply by matthewhisle** [reply]
Awesome thank you for the reply. I could be doing something wrong. I upgraded to 4.8.3, and tried a new device creating database from scratch, but it didn't seem to create the fields I had in the alter section. If you have trouble reproducing this, I could've done something wrong. And seriously thank you for all your help. You are so fast at responding, and this library is awesome! I cannot thank you enough.

**Reply by ujjwalguptaofficial** [reply]
🙏
